### PR TITLE
[gha] enable code coverage when a PR is labeled

### PR DESCRIPTION
**Pull request by sausagee**
### Description
Summary changes to enable code coverage
  - trigger code coverage on PR when the specific label is set 
  - trigger code coverage daily schedule in the main branch
  - increase unit test coverage timeout to 10hrs
  - make llvm-cov verbose for debug
  - cancel concurrent ts sdk e2e tests

### Test Plan
CI triggers the code coverage job if label is set.
- https://github.com/aptos-labs/aptos-core/actions/runs/6669692708/job/18128015111?pr=10684

https://github.com/aptos-labs/aptos-core/pull/10684#issuecomment-1783644552

image

### CODE REVIEW

**Review by brmataptos — APPROVED** [review]

**Review by geekflyer — APPROVED** [review]

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -37,20 +48,20 @@ jobs:
...
           fetch-depth: 0 # get all the history because cargo xtest --change-since origin/main requires it.
       - uses: aptos-labs/aptos-core/.github/actions/rust-setup@main
       - run: rustup component add llvm-tools-preview
-      - uses: taiki-e/install-action@6f1ebcd9e21315fc37d7f7bc851dfcc8356d7da3 # pin@v1.5.6
+      - uses: taiki-e/install-action@f860c89ccbfa08ae7bc92502c27ab631f48b8f9d # pin@v2.21.2
         with:
           tool: nextest,cargo-llvm-cov
       - run: docker run --detach -p 5432:5432 cimg/postgres:14.2
-      - run: cargo llvm-cov --ignore-run-fail --package smoke-test --lcov --output-path lcov_smoke.info
+      - run: cargo llvm-cov --ignore-run-fail --package smoke-test --lcov --output-path lcov_smoke.info -vv
```
Comment (geekflyer) [inline .github/workflows/coverage.yaml:64]: is that -vv intentional?

**Comment by brmataptos** [comment]
I assumed so, given that execution is highly gated and we're trying to
debug the process.

File: .github/workflows/coverage.yaml (hunk 1 above)
Comment (sausagee) [reply .github/workflows/coverage.yaml:64]: Yes, for debugging and for checking the job status. Without the increased verbosity, we don't know if the cmd is running slowly or getting stuck from the CI perspective.

**Comment by sausagee** [comment]
After bumping the install-action to v2.21.2, the smoke test job failed with the following message on the profile data.  
```
2023-10-28T21:01:34.6953691Z      Running `/home/runner/.rustup/toolchains/1.72.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata merge -sparse -f /runner/_work/aptos-core/aptos-core/target/llvm-cov-target/aptos-core-profraw-list -o /runner/_work/aptos-core/aptos-core/target/llvm-cov-target/aptos-core.profdata`
2023-10-28T21:01:46.0337739Z warning: /runner/_work/aptos-core/aptos-core/target/llvm-cov-target/aptos-core-315742-10090495731686504179_0.profraw: invalid instrumentation profile data (file header is corrupt)
2023-10-28T21:01:46.0339940Z error: no profile can be merged
```

I decided to back up the llvm-cov version from this PR and land without it.  This is to unblock what we need for compiler v2 (code coverage).  We can revisit the version update in a separate PR.

**Pull request merged**